File: glowxq-api/business/business-oj/src/main/java/com/glowxq/oj/problem/enums/ProblemSourceType.java
package com.glowxq.oj.problem.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum ProblemSourceType implements BaseType {
    GlowOJ("GlowOJ", "GlowOJ"),
    Hoj("Hoj", "Hoj"),
    Hydro("Hydro", "Hydro"),
    Other("Other", "Other");

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static ProblemSourceType matchCode(String code) {
        for (ProblemSourceType pushStatus : ProblemSourceType.values()) {
            if (pushStatus.getCode().equals(code)) {
                return pushStatus;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/oj/problem/ProblemSourceType.ts
import type { BaseEnum } from '@/enums/base';

// 题目来源类型枚举接口
interface ProblemSourceTypeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * 题目来源类型枚举
 * 用于标识不同的题目来源
 */
export const ProblemSourceType: { [key: string]: ProblemSourceTypeEnum } = {
  /**
   * GlowOJ
   */
  GlowOJ: {
    code: 'GlowOJ',
    name: 'GlowOJ',
    tooltip: 'GlowOJ自建题目',
    text: 'GlowOJ自建题目'
  },

  /**
   * Hoj
   */
  HOJ: {
    code: 'Hoj',
    name: 'Hoj',
    tooltip: 'Hoj',
    text: 'Hoj来源的题目'
  },

  /**
   * Hydro
   */
  HYDRO: {
    code: 'Hydro',
    name: 'Hydro',
    tooltip: 'Hydro',
    text: 'Hydro来源的题目'
  },

  /**
   * Other
   */
  OTHER: {
    code: 'Other',
    name: 'Other',
    tooltip: 'Other',
    text: '其他来源的题目'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): ProblemSourceTypeEnum | null {
  return Object.values(ProblemSourceType).find(item => item.code === code) || null;
}
